File: Cap 12/12.3 - Metodo Externo/cometext.rb
=begin
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.3
 * 
 * Programa ..: USO DE METODO EXTERNO A UMA CLASSE
 * 
=end

class CLS_ALUNO
  attr_accessor :nome, :notas, :media

  def cmedia()
    soma = 0
    for i in 0..3
      soma = soma + notas[i]
    end
    media = soma / 4
    self.media = media 
    return media
  end
end

aluno = CLS_ALUNO.new

puts "DADOS ESCOLARES"
puts

print "Informe o nome: "
aluno.nome = gets.chomp

puts "Informe as notas: "
aluno.notas = []

for i in 0..3
  print "#{i + 1}a. nota: "
  aluno.notas << gets.chomp.to_f
end

aluno.cmedia()

puts
puts "RELATORIO ESCOLAR"
puts
puts "Nome: #{aluno.nome}"
for i in 0..3
  puts "#{i + 1}a. nota: #{aluno.notas[i]}"
end
puts "Media: #{aluno.media}"

File: Cap 12/12.3 - Metodo Externo/cometext.ts
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 12 - Topico 12.3
 * 
 * Programa ..: USO DE METODO EXTERNO A UMA CLASSE
 * 
 */

class CLS_ALUNO {
  NOME: string;
  NOTAS: number[];
  MEDIA: number;

  constructor() {
    this.NOME = "";
    this.NOTAS = [];
    this.MEDIA = 0;
  }
}

function CMEDIA(aluno: CLS_ALUNO): number {
  let SOMA = 0;
  for (let I = 0; I <= 3; I++) {
    SOMA += aluno.NOTAS[I];
  }
  aluno.MEDIA = SOMA / 4;
  return aluno.MEDIA;
}

const ALUNO = new CLS_ALUNO();

console.log("DADOS ESCOLARES\n");

const readline = require('readline-sync');

ALUNO.NOME = readline.question("Informe o nome: ");
console.log("\nInforme as notas: ");

for (let I = 0; I < 4; I++) {
  ALUNO.NOTAS[I] = parseFloat(readline.question(`${I + 1}a. nota: `));
}

CMEDIA(ALUNO);

console.log("\nRELATORIO ESCOLAR\n");

console.log("Nome: " + ALUNO.NOME);
for (let I = 0; I <= 3; I++) {
  console.log(`${I + 1}a. nota: ${ALUNO.NOTAS[I]}`);
}
console.log("Media: " + ALUNO.MEDIA.toFixed(1));
